api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirGlobalPropertyDaoImpl.java

```diff
@@ -47,10 +46,16 @@ public GlobalProperty getGlobalPropertyObject(String property) {
 	}
 	
 	@Override
-	public List<String> getGlobalProperties(String... properties) {
+	public HashMap<String, String> getGlobalProperties(String... properties) {
+		HashMap<String, String> globalPropertiesMap = new HashMap<>();
+		
 		Collection<GlobalProperty> globalProperties = (sessionFactory.getCurrentSession()
 		        .createCriteria(GlobalProperty.class).add(Restrictions.in("property", properties)).list());
-		return globalProperties.stream().filter(p -> p != null).map(GlobalProperty::getPropertyValue)
-		        .collect(Collectors.toList());
+		
+		for (GlobalProperty property : globalProperties) {
+			globalPropertiesMap.put(property.getProperty(), property.getPropertyValue());
+		}
+		
+		return globalPropertiesMap;
```

Comment:
Instead, utilize Stream API 
```java
globalProperties.stream().filter(p -> p != null).map(GlobalProperty::getPropertyValue)
		        .collect(Collectors.toMap(p::getProperty(), p::getPropertyValue()));
```

Reply:
I think in this case, the non-Stream API version is a bit easier to read.